registry/mavrickrishi/modules/aws-ami-snapshot/README.md

```diff
@@ -0,0 +1,197 @@
+
+## Usage
+
+### Basic Usage
+
+```hcl
+module "ami_snapshot" {
+  source = "registry.coder.com/modules/aws-ami-snapshot"
+
+  instance_id     = aws_instance.workspace.id
+  default_ami_id  = data.aws_ami.ubuntu.id
+  template_name   = "aws-linux"
+}
+
+resource "aws_instance" "workspace" {
+  ami           = module.ami_snapshot.ami_id
+  instance_type = "t3.micro"
+
+  # Prevent Terraform from recreating instance when AMI changes
+  lifecycle {
+    ignore_changes = [ami]
+  }
+}
+```
+
+### With Optional Cleanup
+
+```hcl
+module "ami_snapshot" {
+  source = "registry.coder.com/modules/aws-ami-snapshot"
+
+  instance_id               = aws_instance.workspace.id
+  default_ami_id           = data.aws_ami.ubuntu.id
+  template_name            = "aws-linux"
+  enable_dlm_cleanup       = true
+  dlm_role_arn            = aws_iam_role.dlm_lifecycle_role.arn
+  snapshot_retention_count = 5
+
+  tags = {
+    Environment = "development"
+    Project     = "my-project"
+  }
+}
+```
+
+### IAM Role for DLM (Optional)
+
+If using automatic cleanup, create an IAM role for Data Lifecycle Manager:
+
+```hcl
+resource "aws_iam_role" "dlm_lifecycle_role" {
+  name = "dlm-lifecycle-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "dlm.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "dlm_lifecycle" {
+  role       = aws_iam_role.dlm_lifecycle_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSDataLifecycleManagerServiceRole"
+}
+```
+
+## Required IAM Permissions
+
+Users need the following IAM permissions for full functionality:
+
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:CreateImage",
+        "ec2:DescribeImages",
+        "ec2:DescribeInstances",
+        "ec2:CreateTags",
+        "ec2:DescribeTags"
+      ],
+      "Resource": "*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "dlm:CreateLifecyclePolicy",
+        "dlm:GetLifecyclePolicy",
+        "dlm:UpdateLifecyclePolicy",
+        "dlm:DeleteLifecyclePolicy"
+      ],
+      "Resource": "*",
+      "Condition": {
+        "StringEquals": {
+          "dlm:Target": "INSTANCE"
+        }
+      }
+    }
+  ]
+}
+```
+
+## How It Works
+
+1. **Snapshot Creation**: When a workspace transitions to "stop", an AMI snapshot is automatically created (if enabled)
+2. **Tagging**: Snapshots are tagged with workspace name, owner, template, and custom labels
+3. **Snapshot Retrieval**: Available snapshots are retrieved and presented as options for workspace start
+4. **AMI Selection**: The module outputs the appropriate AMI ID (default or selected snapshot)
+5. **Cleanup**: Optional DLM policies can automatically clean up old snapshots
+
+## Variables
```

Comment:
Please delete the Variables and outputs section. We already render variables as a separate tab.